app/src/main/java/com/rob729/quiethours/QuietHoursApp.kt

```diff
@@ -1,10 +1,18 @@
 package com.rob729.quiethours
 
 import android.app.Application
+import androidx.work.Configuration
 import com.rob729.quiethours.util.StoreSession
-class QuietHoursApp : Application() {
+import com.rob729.quiethours.util.WorkManagerHelper
+
+class QuietHoursApp : Application(), Configuration.Provider {
+    override fun getWorkManagerConfiguration() =
```

Comment:
why the Application class is extending Configuration.Provider and overriding getWorkManagerConfiguration()?

Reply:
@rob729 it is required in order to Implement Configuration.Provider, you can visit the following link where it is mentioned:
https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration#implement-configuration-provider

Reply:
If you remove this piece of code from application class then does it work?
I am asking because in the WorkManagerHelper.init() also you are defining val configuration.

Reply:
Yes, you are right. It is not required, the changes I made in the Manifest file were enough to tackle the errors that were coming. I will remove this piece of code.